Tests/AssemblyToProcessTests.cs

```diff
@@ -330,6 +330,81 @@ public void OnPropertyNameChangedMethodWithBeforeAfterIsCalled()
         Assert.DoesNotContain(testResult.Warnings, w => w.Text.ContainsWholeWord(nameof(ClassWithOnChangedBeforeAfter)));
     }
 
+    [Fact]
+    public void OnPropertyNameChangedMethodWithBeforeAfterTypedIsCalled()
+    {
+        var instance = testResult.GetInstance(nameof(ClassWithOnChangedBeforeAfterTyped));
+        instance.Property1 = "foo";
+        instance.Property2 = 1;
+
+        Assert.Equal("-foo", instance.OnProperty1ChangedCalled);
+        Assert.Equal("0-1", instance.OnProperty2ChangedCalled);
+
+        instance.Property1 = "bar";
+        instance.Property2 = 2;
+
+        Assert.Equal("foo-bar", instance.OnProperty1ChangedCalled);
+        Assert.Equal("1-2", instance.OnProperty2ChangedCalled);
+
+        Assert.DoesNotContain(testResult.Warnings, w => w.Text.ContainsWholeWord(nameof(ClassWithOnChangedBeforeAfterTyped)));
+    }
+
+    [Fact]
+    public void OnPropertyNameChangedMethodWithBeforeAfterTypedWithInvalidSignatureDefaultIsNotCalled()
+    {
+        var instance = testResult.GetInstance(nameof(ClassWithOnChangedBeforeAfterTypedInvalidSignatureDefault));
+        instance.Property1 = "foo";
+
+        Assert.Null(instance.OnProperty1ChangedCalled);
+
+        instance.Property1 = "bar";
+
+        Assert.Null(instance.OnProperty1ChangedCalled);
+
+        Assert.DoesNotContain(testResult.Warnings, w => w.Text.ContainsWholeWord(nameof(ClassWithOnChangedBeforeAfterTypedInvalidSignatureDefault)));
```

Comment:
Don't we want a warning to be emitted in this case?

Reply:
Yes, missed that.